core/print_api.php

```diff
@@ -1651,21 +1651,28 @@ function print_email_link( $p_email, $p_text ) {
  * @param string  $p_text   Link text to display to user.
  * @param string  $p_tooltip The tooltip to show.
  * @param string  $p_bug_id The bug identifier.
- * @param boolean $p_show_as_text By default, show link as button with envelope icon.
+ * @param boolean $p_show_as_button If true, show link as button with envelope
+ *                                  icon, otherwise display a plain-text link.
  * @return void
  */
-function print_email_link_with_subject( $p_email, $p_text, $p_tooltip, $p_bug_id, $p_show_as_text = false )
+function print_email_link_with_subject( $p_email, $p_text, $p_tooltip, $p_bug_id, $p_show_as_button = true )
 {
-	if( !is_blank( $p_tooltip ) && $p_tooltip != $p_text ) {
-		$t_tooltip = ' title="' . $p_tooltip . '"';
-	} else {
-		$t_tooltip = '';
-	}
-
+	global $g_project_override;
 	$t_bug = bug_get( $p_bug_id, true );
-	$t_show_user_email_threshold = config_get( 'show_user_email_threshold', null, null, $t_bug->project_id );
+
+	$g_project_override = $t_bug->project_id;
+
+	echo prepare_email_link(
+			$p_email,
+			$p_text,
+			email_build_subject( $p_bug_id ),
+			$p_tooltip,
+			$p_show_as_button
+		);
+	return;
```

Comment:
Why is this `return` statement here?

Reply:
Good catch, thanks. That's a leftover from refactoring, and I forgot to remove the dead code after this return statement. The logic has been moved to other API functions (prepare_email_link(), prepare_mailto_url()). 

I'll update the PR.